src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateEqualsAndGetHashCodeFromMembers/GenerateEqualsAndGetHashCodeFromMembersTests.cs

```diff
@@ -119,7 +306,12 @@ public override bool Equals(object obj)
                EqualityComparer<S>.Default.Equals(a, program.a);
     }
 }",
-parameters: CSharp6Implicit);
+parameters: CSharp6Implicit,
+expectedDiagnostics: new List<DiagnosticResult>
+{
+    // /0/Test0.cs(5,11): error CS0535: 'S' does not implement interface member 'IEquatable<S>.Equals(S)'
+    DiagnosticResult.CompilerError("CS0535").WithSpan(5, 11, 5, 24).WithArguments("S", "System.IEquatable<S>.Equals(S)"),
+});
```

Comment:
diagnostics added where appropraite.  note: these are no cases where the product code is introducing unexpected diagnostics.  rather, these are either diagnostics about the starting state of the code.  Or they are acceptable diagnostics produced by the refactoring.  FOr example, if the user code refers to an unknown type, then `generate equals` will still reference that unknown type, leading to diagnostics in the initial code, and new diagnostics after the refactoring.

Reply:
If the diagnostic is known to be desired for this test, use markup syntax `{|CS0535:text|}` instead of setting `expectedDiagnostics`. Compiler errors in `expectedDiagnostics` represent cases where the test was improperly constructed and/or the code fix is misbehaving.

Reply:
Done.  Thanks!